Mathlib/Tactic/MoveAdd.lean

```diff
@@ -0,0 +1,379 @@
+/-
+Copyright (c) 2023 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino, Damiano Testa
+-/
+import Mathlib.Algebra.Group.Basic
+
+/-!
+
+#  `move_add` a tactic for moving summands in expressions
+
+The tactic `move_add` rearranges summands in expressions.
+It does so by receiving as input a list of "summand atoms", inducing the desired permutation.
+See the doc-string for `move_oper` for details of how the permutation is determined by the input.
+For the technical description, look at `Mathlib.MoveAdd.weight` and `Mathlib.MoveAdd.reorderUsing`.
```

Comment:
I think there needs to be a much clearer and human readable spec of what this is doing. At the moment when I look at the examples I am uniformly confused by the intended semantics.

Reply:
I gave this a shot: now the doc-module contains more details.

Let me know what is still unclear!